src/main/java/com/lob/client/AsyncLobClient.java

```diff
@@ -532,12 +532,16 @@ public void onThrowable(final Throwable t) {
         public T onCompleted(final Response response) throws Exception {
             if (isSuccess(response)) {
                 final T value = MAPPER.readValue(response.getResponseBody(), clazz);
```

Comment:
Like in this function, is there a reason to use generics when `onCompleted` requires it to be an `AbstractResponse`? At the very least, I'd like to see the T in the class signature to extend `AbstractResponse` since it's now kind of a pseudo requirement for T.